src/client/client.ts

```diff
@@ -92,11 +92,17 @@ class ClientRequestImpl {
     setBody = !(methodUpperCase === 'GET' || methodUpperCase === 'HEAD')
 
     // Pass URL string to 1st arg for testing with MSW and node-fetch
-    return (opt?.fetch || fetch)(url, {
+    const res = (opt?.fetch || fetch)(url, {
       body: setBody ? this.rBody : undefined,
       method: methodUpperCase,
       headers: headers,
     })
+
+    if (opt?.after) {
+      const hookedResponse = await opt.after(await res)
```

Comment:
I would also like to receive the request object with a hook. What do you think?

The code is a bit long, but if it looks like the following, we won't have to worry about the order of the parameters if we add retry options later.


```diff
diff --git a/src/client/client.test.ts b/src/client/client.test.ts
index b2c8fde..52e0331 100644
--- a/src/client/client.test.ts
+++ b/src/client/client.test.ts
@@ -384,8 +384,8 @@ describe('After Hook', () => {
   const app = new Hono().get('/hello', (c) => c.text('OK'))
   type AppType = typeof app
   const client = hc<AppType>('http://localhost', {
-    after: (res) => {
-      res.headers.append('X-Custom', 'Hono!')
+    after: ({ response }) => {
+      response.headers.append('X-Custom', 'Hono!')
     },
   })
 
diff --git a/src/client/client.ts b/src/client/client.ts
index 01543b6..2814860 100644
--- a/src/client/client.ts
+++ b/src/client/client.ts
@@ -92,14 +92,18 @@ class ClientRequestImpl {
     setBody = !(methodUpperCase === 'GET' || methodUpperCase === 'HEAD')
 
     // Pass URL string to 1st arg for testing with MSW and node-fetch
-    const res = (opt?.fetch || fetch)(url, {
+    const req = new Request(url, {
       body: setBody ? this.rBody : undefined,
       method: methodUpperCase,
       headers: headers,
     })
+    const res = (opt?.fetch || fetch)(req)
 
     if (opt?.after) {
-      const hookedResponse = await opt.after(await res)
+      const hookedResponse = await opt.after({
+        request: req,
+        response: await res,
+      })
       return hookedResponse ?? res
     }
     return res
diff --git a/src/client/types.ts b/src/client/types.ts
index ca26125..f94ffb1 100644
--- a/src/client/types.ts
+++ b/src/client/types.ts
@@ -12,7 +12,10 @@ type Data = {
   output: {}
 }
 
-type AfterResponseHook = (res: Response) => Promise<Response | void> | Response | void
+type AfterResponseHook = (arg: {
+  request: Request
+  response: Response
+}) => Promise<Response | void> | Response | void
 
 export type RequestOptions = {
   headers?: Record<string, string>

```